core/src/main/java/io/grpc/internal/LogExceptionRunnable.java

```diff
@@ -56,7 +58,8 @@ public void run() {
       task.run();
     } catch (Throwable t) {
       log.log(Level.SEVERE, "Exception while executing runnable " + task, t);
-      throw t instanceof RuntimeException ? (RuntimeException) t : new RuntimeException(t);
+      Throwables.propagateIfPossible(t);
+      throw new AssertionError(t);
```

Comment:
This will either be an Exception or a raw Throwable (or a subclass of Throwable that isn't RE or E).

Is wrapping in an AE appropriate?